package/base-files/files/etc/profile

```diff
@@ -49,6 +49,36 @@ in order to prevent unauthorized SSH logins.
 EOF
 fi
 
+DEFAULT_KEY=$(uci -q get system.@system[0].default_key)
+if [ -n "$DEFAULT_KEY" ] ; then
+	WIFI_LIST=$(uci show wireless | grep $DEFAULT_KEY | cut -d'.' -f 2 | while read IFACE ; do
+		echo -e "\t$(uci get wireless.${IFACE}.ssid) ($(uci get wireless.${IFACE}.device))"
+	done)
+fi
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+BLINK='\033[5m'
+BOLD='\033[1m'
+NC='\033[0m'
+
+[ -n "$WIFI_LIST" ] && echo -e \
+"===$BLINK$RED$BOLD WARNING! $NC=====================================
+${YELLOW}WiFi set with LABEL_MAC unsecure key!
+Some WiFi SSIDs are configured using a default
+key ($DEFAULT_KEY) based on your device's MAC.
+In order to prevent unauthorized WiFi connections,
+please change the key in /etc/config/wireless file
+to the following SSIDs:
+$GREEN$WIFI_LIST$NC
```

Comment:
At least in my opinion, use of escape sequences and colors are problematic in two ways:
* Color as a highlight is a poor choice for those with visual impairments
* Not all "terminals" support escape sequences

Yellow, in particular, is very challenging as it provides very little brightness differentiation from white or tan backgrounds.


If you're going to chew up a bunch of bytes to make the text more "in your face", perhaps something like
```
========================================
===    WARNING: YOUR DEVICE IS NOT SECURE    ===
========================================

Your wireless password is based on your device's MAC address.

It can be read over wireless without logging in to your wireless.

You should IMMEDIATELY review and change your wireless passwords
either using LuCI or by editing /etc/config/wireless and then  # wifi reload
```
The runs of `=` should compress well into the ROM.

_(Also, I couldn't quickly convince myself that $DEFAULT_KEY is always populated.)_

Reply:
Thanks for the comment @jeffsf!
I have a question from my ignorance point of view... when we use the device's MAC we change the BSSID to a random value, how can anybody read the mac?

The yellow color is just for an extra highlight and the warning style itself was mimicking the root password warning, but if consistency with root warning is not a problem, I completely agree with your style. We can change the root password warning style too if you consider it would be better. Let me know your thoughts about it :)

Reply:
@mgiganto The label MAC address is written on the label of the router... If there is neither a label PSK nor a label MAC address it's not possible to get even a little bit of security without using either WPS (support was dropped on many platforms like Android, Linux and Apple) or a trusted third-party (ping me if you want me to explain that approach, but that approach is not an option for OpenWrt I think).

Reply:
> when we use the device's MAC we change the BSSID to a random value, how can anybody read the mac?

My apologies, I had missed the code in `mac80211.sh` at 103 that randomized the _MAC address_ of the wireless adapter.